AdyenCard/Components/GiftCardComponent/GiftCardComponent.swift

```diff
@@ -0,0 +1,304 @@
+    PaymentComponent,
+    PresentableComponent,
+    Localizable,
+    LoadingComponent,
+    Observer {
+
+    /// Indicates Gift card related errors.
+    public enum Error: LocalizedError {
+
+        /// Indicates that the balance check failed for some reason.
+        case balanceCheckFailure
+
+        /// Indicates the payment object passed to the `GiftCardComponent` is nil or invalid.
+        case invalidPayment
+
+        /// Indicates that the `partialPaymentDelegate` is nil.
+        case missingPartialPaymentDelegate
+
+        /// Indicates any other error
+        case otherError(Swift.Error)
+
+        /// :nodoc:
+        public var errorDescription: String? {
+            switch self {
+            case .balanceCheckFailure:
+                return "Due to a network issue we couldn’t retrieve the balance. Please try again."
+            case .invalidPayment:
+                return "For gift card flow to work, you need to provide the Payment object to the component."
+            case .missingPartialPaymentDelegate:
+                return "Please provide a `PartialPaymentDelegate` object"
+            case let .otherError(error):
+                return error.localizedDescription
+            }
+        }
+    }
+
+    /// :nodoc:
+    private var giftCardPaymentMethod: GiftCardPaymentMethod
+
+    /// The gift card payment method.
+    public var paymentMethod: PaymentMethod { giftCardPaymentMethod }
+
+    /// Describes the component's UI style.
+    public let style: FormComponentStyle
+
+    /// The delegate of the component.
+    public weak var delegate: PaymentComponentDelegate?
+
+    /// The partial payment component delegate.
+    public weak var partialPaymentDelegate: PartialPaymentDelegate?
+
+    /// The delegate that handles shopper confirmation UI when the balance of the gift card is sufficient to pay.
+    public weak var readyToSubmitComponentDelegate: ReadyToSubmitPaymentComponentDelegate?
+
+    /// Initializes the card component.
+    ///
+    /// - Parameters:
+    ///   - paymentMethod: The gift card payment method.
+    ///   -  clientKey: The client key that corresponds to the webservice user you will use for initiating the payment.
+    /// See https://docs.adyen.com/user-management/client-side-authentication for more information.
+    ///   -  style: The Component's UI style.
+    public init(paymentMethod: GiftCardPaymentMethod,
+                clientKey: String,
+                style: FormComponentStyle = FormComponentStyle()) {
+        self.giftCardPaymentMethod = paymentMethod
+        self.style = style
+        self.environment.clientKey = clientKey
+    }
+
+    // MARK: - Presentable Component Protocol
+
+    /// :nodoc:
+    public var viewController: UIViewController { securedViewController }
+
+    /// :nodoc:
+    public var requiresModalPresentation: Bool { true }
+
+    private lazy var securedViewController = SecuredViewController(child: formViewController, style: style)
```

Comment:
can we make `viewController` a lazy var and initialize it with `SecuredViewController(child: formViewController, style: style)`?